pkg/alertmanager/operator.go

```diff
@@ -1074,10 +1074,10 @@ func checkWechatConfigs(ctx context.Context, configs []monitoringv1alpha1.WeChat
 	for i, config := range configs {
 		wechatConfigKey := fmt.Sprintf("%s/wechat/%d", key, i)
 
-		if config.APIURL != nil {
-			_, err := url.Parse(*config.APIURL)
+		if len(config.APIURL) > 0 {
+			_, err := url.Parse(config.APIURL)
```

Comment:
I guess this is a validation we cannot do via the CRD itself? If its a valid URL? (Sorry if I missed it somewhere)

Reply:
AFAICT yes, you can't really do complex validation like in the API. Also the client is free to skip API validations (`kubectl apply --validate=false -f ...`) so we still need to check it in the controller's code.